pkg/safeguards/safeguards.go

```diff
@@ -68,23 +64,40 @@ func GetManifestViolations(ctx context.Context, manifestFiles []string) ([]Manif
 		return manifestViolations, err
 	}
 
+	// organized map of manifest object by file name
+	manifestMap := make(map[string][]*unstructured.Unstructured, 0)
+
+	// aggregate of every manifest object into one list
+	objects := []*unstructured.Unstructured{}
+
 	for _, m := range manifestFiles {
-		var objectViolations map[string][]string
-		objects, err := fc.ReadManifests(m) // read all the objects stored in a single file
+		objs, err := fc.ReadManifests(m.Path) // read all the objects stored in a single file
```

Comment:
nit: having objs and objects in the same scope could benefit from more precise variable names

Reply:
fixed